src/components/CreateVmWizard/steps/BasicSettings.js

```diff
@@ -289,6 +305,25 @@ class BasicSettings extends React.Component {
         }
         break
 
+      case 'enableSysPrepTimezone': // Configure Timezone checkbox change
+        changes[field] = value
+        // if the Configure Timezone checkbox checked, sysprep timezone should be set, otherwise reset it to the default/no value
+        changes.initTimezone = value ? this.props.data.initTimezone || this.props.data.selectedSysprepTimezone : ''
+        break
+
+      case 'initTimezone': // sysprep timezone change
+        changes[field] = value
+        changes.selectedSysprepTimezone = value // save the last selected sysprep timezone which is not always the same as sysprep timezone being actually set
```

Comment:
I am not sure. the word "last" can be confusing, especially in the case when sysprep enabled because in this case it represents the **actual** selected sysprep timezone.

Reply:
It sees to me that the true actual selected value is `changes.initTimezone`. This is what the outside world sees and what is presented in the summary step. This is also consistent with other props i.e. `initHostname', `initSshKeys`, ...
Not displaying this value to the user exposes us to bugs caused by synchronization - user sees different props then actually used to create vm (see the bug scenarios my comments).
Logically `selectedSysprepTimezone` is just a special case of default value (last selected value) so I would rename it and limit it uses only to this role.

Reply:
I have no problem with renaming the variable, just thinking about more suitable name as it has now.

> It seems to me that the true actual selected value is changes.initTimezone. This is what the outside world sees and what is presented in the summary step.

This is true and it should be true in case when Windows OS selected and sysprep enabled. I don't see any problem with that. The line above is related to the situation when user changes sysprep timezone. Also the comment says about that. Also note that `selectedSysprepTimezone` is not always the same as sysprep timezone being actually set (`initTimezone`). For example in the case the user disables sysprep or configuring timezone for sysprep which is enabled.

> Not displaying this value to the user exposes us to bugs caused by synchronization - user sees different props then actually used to create vm (see the bug scenarios my comments).

I'm not exactly sure what you mean by this but `selectedSysprepTimezone` should be visible in the Basic settings, and only if Windows OS selected and sysprep enabled. And the same value should be visible in the Review step but only in this case. And I haven't found any issue regarding inconsistency of this value in Basic settings and Review step.

Regarding the bug scenario you've found, there are more bugs. I was investigating it and one of the problems is missing logic for cluster changes but also some logic for `timeZone` and `operatingSystemId` is missing, too - and this isn't quite related to this PR, but I can add it anyway. The second problem is related to the cluster CPU type chosen in webadmin when creating a cluster. Looks like that it works fine in the Basic settings only when selecting clusters with CPU Types available for CPU architecture x86_64. I'll create a separate issue for that (if it doesn't exist yet). It's present on our master.

Reply:
> selectedSysprepTimezone is more accurate 

It could be more accurate but we would need to refactor much more code to make it consistent.
Currently clearly we name all sysprep props using `init` prefixes. Which is motivated by re-using the same UI.
Having one field (partially) using different naming is confusing.

> there are more bugs

Yep. We are in callback hell here. However both bug scenario are caused by setting default value (otherwise there would be empty value which is OK)
So they are related to this PR.

Reply:
I was investigating more [this bug scenario](https://github.com/oVirt/ovirt-web-ui/pull/1288#discussion_r503230034) you've found. There is **missing logic for properly changing the OS id with the change of the cluster**. The OS id remains the same as in step 1, but as the `operatingSystemList` has changed with the change of the cluster, we cannot find the actual OS id in that list of the operating systems => blank OS in the drop down. And no change of the OS id => no change in the sysprep section, logically. I've implemented cloud/init sysprep related changes everywhere in the `handleChange` where the OS id was changed. I'm not going to fix this in this PR. This problem will be fixed separately by https://github.com/oVirt/ovirt-web-ui/issues/1307.